t/01_style.t

```diff
@@ -12,6 +13,17 @@ ok system(qq{git grep -I -l '[#/ ]*SPDX-License-Identifier ' ':!t/01_style.t'})
 $out = qx{git grep -ne "check_var('ARCH',.*)" -e "check_var('BACKEND',.*)" ':!lib/Utils/Architectures.pm' ':!lib/Utils/Backends.pm' 'lib' 'tests'};
 ok $? != 0 && $out eq '', 'No check_var function to verify ARCH/BACKEND types' or diag $out;
 ok system(qq{git grep -I -l \\( -e "egrep" -e "fgrep" \\) ':!t/01_style.t' ':!CONTRIBUTING.md'}) != 0, 'No usage of the deprecated egrep and fgrep commands';
-$out = qx{git grep -I -l 'nots3cr3t' ':!data/wsl/Autounattend_*.xml' 'data' | xargs grep -E -L '(luks|encryption).*password.*nots3cr3t'};
-ok $? != 0 && $out eq '', 'No plain password on data directory' or diag $out;
+
+# Find all files in 'data' directory containing 'nots3cr3t', excluding specific WSL files.
+# Then, filter out files that have an allowed context for the password.
+my @bad_files = grep {
+    chomp $_;    #  chomp $_ the filename
+                 # Allowed pattern: file contains 'nots3cr3t' but on the same line of other keywords.
+                 # If the file contains the secret but does NOT contain the allowed pattern, it's problematic
+    path($_)->slurp !~ qr/(luks|encryption).*password.*nots3cr3t/;
+} qx{git grep -I -l 'nots3cr3t' ':!data/wsl/Autounattend_*.xml' 'data'};
+# As grep {} expect a list, qx doesn't return the command's output as a single string, but is returns
+# a list of strings, where each element is one line from the command's output.
+ok !@bad_files, 'No plain password in files within data directory';
+
```

Comment:
We managed to have rather simple one-line checks. Obviously the previous change in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/21687/files#diff-a753415a95b42ddfd0bf68f9ed52689b778fc3eec71ce0b7134e07b0507d46edR15 was cynical and introducing egrep when just the line in before checks for no use of egrep. But this was fixed with the rather portable call `grep -E` which we can rely on for the foreseeable future. This "native" implementation is not exactly easier to read. So, sorry, I prefer the previous version before this PR

Reply:
I described the issue in more details in https://progress.opensuse.org/issues/185146

Reply:
No, I think it is fine it passing.